include/exercise_functions.php

```diff
@@ -0,0 +1,472 @@
+<?php
+
+function addExercise(){
+
+  $exerciseTypeId = getExerciseTypeId();
+
+  $exerciseName = $_REQUEST["exerciseName"];
+  $session = $_REQUEST["session"];
+  $exerciseTarget = $_REQUEST["target"];
+  $userId = $_SESSION['userId'];
+
+  db_Query("
+    INSERT INTO exercise(exerciseName, typeId, session, target, userId)
+    VALUES(:exerciseName, :typeId, :session, :target, :userId)
+  ",
+  
+  [
+    'exerciseName' => $exerciseName,
+    'typeId' => $exerciseTypeId,
+    'session' => $session,
+    'target' => $exerciseTarget,
+    'userId' => $userId
+  ]
+
+  
+  );
+}
+
+function getExerciseTypes(){
+
+  $exerciseTypeData = db_Query("
+    SELECT *
+    FROM exercise_type
+  ")->fetchAll();
+
+  return $exerciseTypeData;
+
+}
+
+function getExerciseTypeId(){
+
+  $exerciseTypeName = $_REQUEST["exerciseType"];
+
+  $exerciseTypeId = db_Query("
+    SELECT typeId
+    FROM exercise_type
+    WHERE typeName = '$exerciseTypeName'
+  ")->fetch();
+
+  return $exerciseTypeId['typeId'];
+
+}
+
+function getExerciseTypeName($individualExercise){
+
+  $typeId = $individualExercise['typeId'];
+
+  $exerciseTypeName = db_Query("
+    SELECT typeName
+    FROM exercise_type
+    WHERE typeId = $typeId
+  ")->fetch();
+
+  return $exerciseTypeName['typeName'];
+
+}
+
+function displayExerciseTypeOptions(){
+
+  $exerciseTypeData = getExerciseTypes();
+    
+
+  foreach($exerciseTypeData as $individualExerciseType){
+
+    $typeId = $individualExerciseType['typeId'];
+    
+    echo"
+      <option value=".$individualExerciseType['typeName'].">".$individualExerciseType['typeName']."</option>
+    ";
+
+  }
+
+}
+
+function getExercises($user){
+
+  $userId = $user['userId'];
+
+  $exerciseData = db_Query("
+    SELECT *
+    FROM exercise
+    WHERE userId = '$userId'
+  ")->fetchAll();
+
+  return $exerciseData;
+
+}
+
+function getSingleExercise($exerciseIdPass){
+
+  $exerciseData = db_Query("
+    SELECT *
+    FROM exercise
+    WHERE exerciseId = '$exerciseIdPass' 
+  ")->fetch();
+
+  return $exerciseData;
+
+}
+
+function getSessions(){
+
+  $sessionData = db_Query("
+    SELECT *
+    FROM session
+  ")->fetchAll();
+
+  return $sessionData;
+
+}
+
+function displayAllExercises($user){
+
+  $allExercises = getExercises($user);
+  
+  foreach($allExercises as $individualExercise){
+    $exerciseTypeName = getExerciseTypeName($individualExercise);
```

Comment:
I would get the exerciseTypeName as part of `getExercises` so that you're not having to do a whole bunch of separate queries. You can do this with a SQL `JOIN` you can include the `typeName` column from the `exercise_type` so that each `individualExercise` here already has that information